src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java

```diff
@@ -281,6 +281,40 @@ LEFT JOIN (SELECT DISTINCT dataset_id AS id FROM dar_dataset) dar_ds_ids ON dar_
     """)
     List<Dataset> getActiveDatasets();
 
+    @UseRowReducer(DatasetReducer.class)
+    @SqlQuery("""
+        SELECT d.dataset_id, d.name, d.create_date, d.create_user_id, d.update_date,
+            d.update_user_id, d.object_id, d.active, d.dac_id, d.alias, d.data_use, d.dac_approval,
+            dar_ds_ids.id AS in_use,
+            u.user_id AS u_user_id, u.email AS u_email, u.display_name AS u_display_name,
+            u.create_date AS u_create_date, u.email_preference AS u_email_preference,
+            u.institution_id AS u_institution_id, u.era_commons_id AS u_era_commons_id,
+            k.key, dp.property_value, dp.property_key, dp.property_type, dp.schema_property, dp.property_id,
+            ca.consent_id, c.translated_use_restriction,
+            fso.file_storage_object_id AS fso_file_storage_object_id,
+            fso.entity_id AS fso_entity_id,
+            fso.file_name AS fso_file_name,
+            fso.category AS fso_category,
+            fso.gcs_file_uri AS fso_gcs_file_uri,
+            fso.media_type AS fso_media_type,
+            fso.create_date AS fso_create_date,
+            fso.create_user_id AS fso_create_user_id,
+            fso.update_date AS fso_update_date,
+            fso.update_user_id AS fso_update_user_id,
+            fso.deleted AS fso_deleted,
+            fso.delete_user_id AS fso_delete_user_id
+        FROM dataset d
+        LEFT JOIN users u on d.create_user_id = u.user_id
+        LEFT JOIN (SELECT DISTINCT dataset_id AS id FROM dar_dataset) dar_ds_ids ON dar_ds_ids.id = d.dataset_id
+        LEFT JOIN dataset_property dp ON dp.dataset_id = d.dataset_id
+        LEFT JOIN dictionary k ON k.key_id = dp.property_key
+        LEFT JOIN consent_associations ca ON ca.dataset_id = d.dataset_id
+        LEFT JOIN file_storage_object fso ON fso.entity_id = d.dataset_id::text AND fso.deleted = false
+        LEFT JOIN consents c ON c.consent_id = ca.consent_id
+        WHERE (dp.schema_property = 'studyName' AND UPPER(dp.property_value) = UPPER(:name))
+    """)
+    Dataset findDatasetByStudyName(@Bind("name") String name);
```

Comment:
There can be many datasets with the same study name. When we parse an incoming dataset submission form, if there are N consent groups, there will be N datasets and each one will have the same study name.

Reply:
Oh interesting. I can make this return a list of ids instead. It seems somewhat unintuitive to me that the studyName property is how multiple datasets are linked as being under the same submission, and that the linking is unique. But trying to envision another approach makes things super complicated so it makes sense.

Reply:
Yeah, I'm thinking that we might want to trim this feature down a bit. The only use case here is to ensure unique study names. If we instead returned just that ... a list of unique study names, we'd speed up the query quite a bit. Separately, the the UI could client-side filter to ensure uniqueness. Just a suggestion - making this return a list is just as functional.

Reply:
Hmm... I feel like returning a list of IDs might be simpler. It could also just return a `true`/`false` -- all it would have to do is check if there is a dataset property with `studyName` of the provided input, which would make it fast. Let me know what you think, and I can implement it either way